packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -514,9 +493,20 @@ export function typeDefinition<T>(
 		for (const def of node.validDefinitions) {
 			if (!attached.has(node.node.originalNode)) {
 				options.attachTypeInfo(node.node.originalNode, def.typeDef)
+				handleAttributes(
+					def.typeDef.attributes,
+					options.context,
+					(handler, c) => {
+						const attacher = handler.attachString?.(c, options.context)
+						if (attacher) {
+							options.stringAttacher(node.node.originalNode, attacher)
```

Comment:
I'm not sure this is the correct approach as this is problematic for multiple valid definitions. I suppose we could just say calling this function multiple times is okay and let the handler deal with it in some way?

Not sure how to handle this though, it doesn't really make sense to have multiple string parsers at the same time, maybe it would be best to just not do anything at all when there are multiple conflicting valid definitions that have an attachString method defined?

Reply:
The attached set makes sure it only runs once. In #1293 I added a comment, but I don't think it makes much sense to run multiple string parsers on the same node. It means you're probably missing a required type used by a dispatcher.